controllers/drpolicy_controller.go

```diff
@@ -0,0 +1,282 @@
+		return ctrl.Result{}, err
+	}
+
+	if mirrorPeer.Spec.Type == multiclusterv1alpha1.Async {
+		clusterFSIDs := make(map[string]string)
+		klog.Infof("Fetching clusterFSIDs")
+		err = r.fetchClusterFSIDs(ctx, mirrorPeer, clusterFSIDs)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return ctrl.Result{RequeueAfter: 60 * time.Second}, nil
+			}
+			return ctrl.Result{}, fmt.Errorf("an unknown error occured while fetching the cluster fsids, retrying again: %v", err)
+		}
+
+		err = r.createOrUpdateManifestWorkForVRC(ctx, mirrorPeer, &drpolicy, clusterFSIDs)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to create VolumeReplicationClass via ManifestWork: %v", err)
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *DRPolicyReconciler) createOrUpdateManifestWorkForVRC(ctx context.Context, mp *multiclusterv1alpha1.MirrorPeer, dp *ramenv1alpha1.DRPolicy, clusterFSIDs map[string]string) error {
+
+	var fsids []string
+	for _, v := range clusterFSIDs {
+		fsids = append(fsids, v)
+	}
+
+	// To ensure reliability of hash generation
+	sort.Strings(fsids)
+
+	replicationId := utils.CreateUniqueReplicationId(fsids)
+
+	for _, pr := range mp.Spec.Items {
+		manifestWorkName := fmt.Sprintf("%s-vrc", pr.ClusterName)
+		found := &workv1.ManifestWork{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      manifestWorkName,
+				Namespace: pr.ClusterName,
+			},
+		}
+
+		err := r.HubClient.Get(ctx, types.NamespacedName{Name: found.Namespace, Namespace: pr.ClusterName}, found)
+
+		switch {
+		case err == nil:
+			klog.Infof("%s already exists. updating...", manifestWorkName)
+		case !errors.IsNotFound(err):
+			klog.Error(err, "failed to get ManifestWork: %s", manifestWorkName)
+			return err
+		}
+		interval := dp.Spec.SchedulingInterval
+		params := make(map[string]string)
+		params[MirroringModeKey] = DefaultMirroringMode
+		params[SchedulingIntervalKey] = interval
+		params[ReplicationSecretNameKey] = RBDReplicationSecretName
+		params[ReplicationSecretNamespaceKey] = pr.StorageClusterRef.Namespace
+		vrcName := fmt.Sprintf(RBDVolumeReplicationClassNameTemplate, utils.FnvHash(interval))
+		labels := make(map[string]string)
+		labels[fmt.Sprintf(RamenLabelTemplate, ReplicationIDKey)] = replicationId
+		vrc := replicationv1alpha1.VolumeReplicationClass{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "VolumeReplicationClass", APIVersion: "replication.storage.openshift.io/v1alpha1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   vrcName,
+				Labels: labels,
+			},
+			Spec: replicationv1alpha1.VolumeReplicationClassSpec{
+				Parameters:  params,
+				Provisioner: fmt.Sprintf(RBDProvisionerTemplate, pr.StorageClusterRef.Namespace),
+			},
+		}
+
+		objJson, err := json.Marshal(vrc)
+		if err != nil {
+			return fmt.Errorf("failed to marshal %v to JSON, error %w", vrc, err)
+		}
+
+		manifest := workv1.Manifest{}
+		manifest.RawExtension = runtime.RawExtension{Raw: objJson}
```

Comment:
This will override existing VRC, won't it? What happens to VRC that is not in ManifestWork?